src/Compilation.zig

```diff
@@ -855,14 +855,31 @@ pub fn create(gpa: Allocator, options: InitOptions) !*Compilation {
             break :blk false;
         };
 
-        const sysroot = blk: {
-            if (options.sysroot) |sysroot| {
-                break :blk sysroot;
-            } else if (options.native_darwin_sdk) |sdk| {
-                break :blk sdk.path;
-            } else {
-                break :blk null;
+        const darwin_native = (comptime builtin.target.isDarwin()) and options.target.isDarwin() and
```

Comment:
This is incorrect logic for `darwin_native`. When `options.is_native_os` is false, `darwin_native` must be false. Otherwise, it's going to look at the system sometimes when it should be cross-compiling. Zig should not rely on the Darwin SDK when cross-compiling.

Reply:
Ah, fair point. The reason I re-introduced `comptime` check is because it then emits darwin-specific code that requires a host that spawn subprocesses which isn't always the case plus it should even exist unless it's a native OS AND Darwin. Can I make it more verbose and add `is_native_os` here but leave the `comptime` check? Or perhaps you have a nicer suggestion for this?